### PR TITLE
WIP arduino core upgrade.

### PR DESCRIPTION
**Pull request by dasj19**
###### Motivation for this change
I would like to upgrade to latest arduino IDE but it still complains about missing libraries...

`BUILD FAILED
/build/source/build/build.xml:134: The following error occurred while executing this line:
/build/source/build/build.xml:638: The following error occurred while executing this line:
/build/source/build/build.xml:220: The following error occurred while executing this line:
/build/source/build/build.xml:887: Could not find file /build/source/build/shared/WiFi101-Updater-ArduinoIDE-Plugin-0.10.10.zip to generate checksum for.`

@cc bjornfor gazally bergey binarin worldofpeace
    robberer matthewbauer
    pSuborbekk jwiegley Ericson2314 witchof0x20 domenkozar cko auntieNeo abbradar

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/arduino/arduino-core/default.nix [hunk 1]
```diff
@@ -116,7 +116,7 @@ stdenv.mkDerivation rec {
 
     # Deliberately break build.xml's download statement in order to cause
     # an error if anything needed is missing from download.nix.
-    substituteInPlace build/build.xml --replace "get src" "get error"
+    #substituteInPlace build/build.xml --replace "get src" "get error"
```
Comment (prusnak) [inline pkgs/development/arduino/arduino-core/default.nix:119]: Why is this commented out?

File: pkgs/development/arduino/arduino-core/downloads.nix [hunk 2]
```diff
@@ -69,28 +69,28 @@
     sha256 = "1dflfrg38f0312nxn6wkkgq1ql4hx3y9kplalix6mkqmzwrdvna4";
   };
   "build/Mouse-1.0.1.zip" = fetchurl {
-    url = "https://github.com/arduino-libraries/Mouse/archive/1.0.1.zip";
+    url = "https://github.com/arduino- libraries/Mouse/archive/1.0.1.zip";
```
Comment (prusnak) [inline pkgs/development/arduino/arduino-core/downloads.nix:72]: Remove the space you added

File: pkgs/development/arduino/arduino-core/downloads.nix [hunk 3]
```diff
@@ -100,10 +100,6 @@
     url = "https://downloads.arduino.cc/liblistSerials/liblistSerials-1.4.2.zip";
     sha256 = "1p58b421k92rbgwfgbihy0d04mby7kfssghpmjb4gk9yix09za3m";
   };
-  "build/shared/WiFi101-Updater-ArduinoIDE-Plugin-0.10.6.zip" = fetchurl {
```
Comment (prusnak) [inline pkgs/development/arduino/arduino-core/downloads.nix:103]: Why did you remove this plugin instead of upgrading it?

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by dasj19** [comment]
I fixed all the small typos and comments and I end with the following:
`BUILD FAILED
/build/source/build/build.xml:134: The following error occurred while executing this line:
/build/source/build/build.xml:638: The following error occurred while executing this line:
/build/source/build/build.xml:220: The following error occurred while executing this line:
/build/source/build/build.xml:869: The following error occurred while executing this line:
/build/source/build/build.xml:873: get doesn't support the "error" attribute
`

**Comment by prusnak** [comment]
You have not pushed your changes with the fixes, please do so.

File: pkgs/development/arduino/arduino-core/downloads.nix [hunk 4]
```diff
@@ -76,35 +76,36 @@
...
-    url = "https://github.com/arduino-libraries/WiFi101-FirmwareUpdater-Plugin/releases/download/v0.10.6/WiFi101-Updater-ArduinoIDE-Plugin-0.10.6.zip";
-    sha256 = "1k23xyr5dmr60y8hb9x24wrgd4mfgvrzky621p6fvawn5xbdq8a3";
+  "build/WiFi101-Updater-ArduinoIDE-Plugin-0.10.10.zip" = fetchurl {
+    url = "https://github.com/arduino-libraries/WiFi101-FirmwareUpdater-Plugin/archive/v0.10.10.zip";
+    sha256 = "1j6xy7ffrapk0brcfknaj72pvmzrf19sks8pm900ad7w897mksmv";
+  };
+   "build/libastylej-2.05.1-4.zip" = fetchurl {
```
Comment (prusnak) [inline pkgs/development/arduino/arduino-core/downloads.nix:103]: This was moved from its original location for no good reason. Move it back where it was - just before "build/liblistSerials"

Also update to `libastylej-2.05.1-5` - you missed this update.

**Review by prusnak — CHANGES_REQUESTED** [review]
You missed some of the updates, that's why the build is failing:

`appbundler-1.0ea-arduino5` and maybe others, please go through `build/build.xml` and double-check if there is something else to fix.

File: pkgs/development/arduino/arduino-core/downloads.nix (hunk 4 above)
Comment (dasj19) [reply pkgs/development/arduino/arduino-core/downloads.nix:99]: I updated the build package for the others archs.... I have to admit that I am a bit los as to how I would continue... Idk how to browse downloads.arduino.cc FTP style... so I would not have to hunt down package names and version numbers

**Comment by prusnak** [comment]
There was much more that had to be done that it was easier to open a new pull request https://github.com/NixOS/nixpkgs/pull/87614

Thank you for your work @dasj19 and you are more than welcome to test and review the other PR/!
